tests/integration/query_helpers/test_query_docs.py

```diff
@@ -25,17 +23,11 @@ def setup_faiss():
     ]
     embeddings = HuggingFaceEmbeddings(model_name="all-MiniLM-L6-v2")
 
-    file_path = "./index/faiss_index"
-
-    if os.path.exists(file_path):
-        db = FAISS.load_local(file_path, embeddings)
-    else:
-        db = FAISS.from_documents(list_of_documents, embeddings)
+    db = FAISS.from_documents(list_of_documents, embeddings)
     yield db
-    db.save_local(file_path)
```

Comment:
@mpaulgreen @onmete  i think the "compromise" solution (based on your discussion above) would be to change this logic to only save the db if we didn't load the file from disk in the first place.

but I also think it's a premature optimization that will lead to inconsistency when someone has a stale cache on disk (which is perhaps the issue @onmete hit?).  To do caching of the embeddings correctly, we would need some sort of hash to ensure that if someone updates the test documents, the local disk file is recreated rather than reused.

in my unscientific testing, running "make test-integration" with no pre-existing index took 33.37s and running it again immediately after (thus presumably reusing the index) took 34.09s.  Which means at best any savings from reusing it is outstripped by other randomness in how long the tests take.  Certainly doesn't seem worth the effort of building a correct hash-based system for caching this content right now, so i'm going to approve this PR.

If in the future we feel that the embedding generation cost is becoming a problem we can revisit this, but again we need to implement it in a way that versions/hashes the local cache so we don't reuse it when it's outdated/stale.